### PR TITLE
Restrict draw apertures to solid circles

### PR DESCRIPTION
**Pull request by dsafdsaf132**
## Summary
- mark solid standard circle apertures during aperture parsing
- create D01 draw and arc geometry only for solid circle standard apertures
- make zero-length solid circle draws render as a single flash image
- add tests for non-circle, holed circle, macro circle, solid circle, and zero-length draw behavior

## Tests
- cargo test --manifest-path wasm/Cargo.toml
- wasm-pack build wasm --target web --out-dir pkg --release
- git diff --check

### CODE REVIEW

File: wasm/src/parser/geometry.rs [hunk 1]
```diff
@@ -1081,22 +1081,25 @@ pub fn execute_interpolation(
     let start_x = state.x;
     let start_y = state.y;
 
-    // Get current aperture
-    if let Some(_aperture) = apertures.get(&state.current_aperture) {
+    // RS-274X draw and arc objects can only be created with a solid standard circle aperture.
+    if let Some(aperture) = apertures.get(&state.current_aperture) {
+        if !aperture.is_solid_circle {
+            return Ok(());
+        }
```
[the remark this answers is not in the document]
Comment (dsafdsaf132) [reply wasm/src/parser/geometry.rs]: Handled in the latest push: zero-length D01 now flashes before the solid-circle draw/arc validation, and I added a rectangle regression test for the non-solid aperture case.

File: wasm/src/parser/geometry.rs [hunk 2]
```diff
@@ -1107,105 +1105,144 @@ pub fn execute_interpolation(
...
+                                sr_start_x,
+                                sr_start_y,
+                                state.layer_scale,
+                                state.mirror_x,
+                                state.mirror_y,
+                                state.layer_rotation,
                             )?;
-                            primitives.extend(line_triangles);
+                            continue;
+                        }
```
[the remark this answers is not in the document]
Comment (dsafdsaf132) [reply wasm/src/parser/geometry.rs]: Handled in the latest push: arc interpolation now only treats equal start/end as a flash when there is no I/J center offset, so full-circle arcs continue through arc generation. I added regression coverage for solid-circle full-circle arcs and for non-solid full-circle arcs remaining non-image.
